web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -263,68 +274,180 @@ public void updateUsersInVirtualStudy(@PathVariable SessionType type, @PathVaria
 
 			HttpEntity<String> httpEntity = new HttpEntity<String>(query, getHttpHeaders());
 			
-            ResponseEntity<List<VirtualStudy>> responseEntity = restTemplate.exchange(
+            ResponseEntity<List<Session>> responseEntity = restTemplate.exchange(
                     sessionServiceURL + SessionType.group + "/query/fetch", HttpMethod.POST, httpEntity,
-                    new ParameterizedTypeReference<List<VirtualStudy>>() {
+                    new ParameterizedTypeReference<List<Session>>() {
                     });
 
-            List<VirtualStudy> virtualStudies = responseEntity.getBody();
+            List<Session> virtualStudies = responseEntity.getBody();
 
             return new ResponseEntity<>(virtualStudies, HttpStatus.OK);
 
 		}
 		return new ResponseEntity<>(HttpStatus.SERVICE_UNAVAILABLE);
 	}
 	
+    @RequestMapping(value = "/settings", method = RequestMethod.POST)
```

Comment:
The way the /settings is set up seems to work for all pages, but the data is stored with StudyViewSetting model. It would be quite difficult to convert if later on we want to add settings for other pages

Reply:
there is `SessionPage` attribute in `PageSettingsData` which identifies what page it is